api.d.ts

```diff
@@ -1,80 +1,115 @@
 /********************************************************
- JobRouter JavaScript-API Stubs v5.1
+ JobRouter JavaScript-API Stubs v2022.1
  ********************************************************/
 type CustomErrorHandler = (errorMessage: string) => void;
-type ElementId = string;
+type DecimalSeparator = "" | "," | ".";
+type DialogFunctionUserParameters = {
+    [key: string]: unknown;
+};
+type DialogSuccessReturnObject = {
+    status: string;
+    result?: {
+        [key: string]: unknown;
+    };
+};
+type DialogErrorReturnObject = {
+    status: string;
+    message?: string;
+    result?: {
+        [key: string]: unknown;
+    };
+};
+type ElementValue = string | number | Date | boolean;
+type MessageData = {
+    [key: string]: ElementValue;
+};
 type StepAction = "abort" | "answer" | "assign" | "back" | "finish" | "jumpTo" | "request" | "resubmission" | "save" | "send";
-type TimeUnit = "s" | "m" | "h" | "d";
 type Stamp = {
-    text: string
-    includeTime: boolean
-    includeDate: boolean
-    color: string
-    onStamp?: ((stamp: Stamp, x: number, y: number) => void)
+    text: string;
+    includeTime: boolean;
+    includeDate: boolean;
+    color: string;
+    onStamp?: (stamp: Stamp, x: number, y: number) => void;
 };
+type StampOptions = {
```

Comment:
The Stamp and StampOptions type have a lot in common. Maybe we could use a combined type for Stamp to avoid the redundance. 
Like `type Stamp = { text: string } & StampOptions`

Reply:
Done in https://github.com/jobrouter/javascript-api/pull/4/commits/253d2ce7fe9045f52d93ef3abead156924668c94